exchange-wrapper/contracts/ExchangeWrapper.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@rarible/exchange-v2/contracts/lib/LibTransfer.sol";
+import "@rarible/exchange-v2/contracts/lib/BpLibrary.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@rarible/exchange-interfaces/contracts/IWyvernExchange.sol";
+import "@rarible/exchange-interfaces/contracts/IExchangeV2.sol";
+import "@rarible/royalties/contracts/LibPart.sol";
+
+
+contract ExchangeWrapper is OwnableUpgradeable {
+    using LibTransfer for address;
+    using BpLibrary for uint;
+    using SafeMathUpgradeable for uint;
+
+
+    IWyvernExchange public wyvernExchange;
+    IExchangeV2 public exchangeV2;
+
+    enum Markets {
+        ExchangeV2,
+        WyvernExchange
+    }
+
+    struct TradeDetails {
+        Markets marketId; //if 1 - market is IWyvernExchange, 0 - market is IExchangeV2
+        uint256 amount;
+        bytes tradeData;
+    }
+
+    function __ExchangeWrapper_init(
+        IWyvernExchange _wyvernExchange,
+        IExchangeV2 _exchangeV2
+    ) external initializer {
+        __Context_init_unchained();
+        __Ownable_init_unchained();
+        wyvernExchange = _wyvernExchange;
+        exchangeV2 = _exchangeV2;
+    }
+
+    function setWyvern(IWyvernExchange _wyvernExchange) external onlyOwner {
+        wyvernExchange = _wyvernExchange;
+    }
+
+    function setExchange(IExchangeV2 _exchangeV2) external onlyOwner {
+        exchangeV2 = _exchangeV2;
+    }
+
+    function singleTransfer(TradeDetails memory tradeDetails, uint[] memory fees) external payable {
```

Comment:
why is the function called transfer? Should it be named singlePurchase?
the same applies to bulk

Reply:
Also, there is a word already used - "trade". Why don't you use the same word everywhere?
I think, purchase is better here. So, it's better to be consistent and use word "purchase" everywhere

Reply:
agree, done